diff --git a/devtools/client/debugger/src/utils/prefs.js b/devtools/client/debugger/src/utils/prefs.js
--- a/devtools/client/debugger/src/utils/prefs.js
+++ b/devtools/client/debugger/src/utils/prefs.js
@@ -38,6 +38,7 @@ if (isDevelopment()) {
   pref("devtools.debugger.start-panel-size", 300);
   pref("devtools.debugger.end-panel-size", 300);
   pref("devtools.debugger.tabsBlackBoxed", "[]");
+  pref("devtools.debugger.wasm-max-lines", 1000000);
   pref("devtools.debugger.ui.editor-wrapping", false);
   pref("devtools.debugger.ui.framework-grouping-on", true);
   pref("devtools.debugger.pending-selected-location", "{}");
@@ -106,6 +107,7 @@ export const prefs = new PrefsHelper("de
   endPanelCollapsed: ["Bool", "debugger.end-panel-collapsed"],
   startPanelSize: ["Int", "debugger.start-panel-size"],
   endPanelSize: ["Int", "debugger.end-panel-size"],
+  wasmMaxLines: ["Int", "devtools.debugger.wasm-max-lines"],
   frameworkGroupingOn: ["Bool", "debugger.ui.framework-grouping-on"],
   pendingSelectedLocation: ["Json", "debugger.pending-selected-location", {}],
   expressions: ["Json", "debugger.expressions", []],
diff --git a/devtools/client/debugger/src/utils/wasm.js b/devtools/client/debugger/src/utils/wasm.js
--- a/devtools/client/debugger/src/utils/wasm.js
+++ b/devtools/client/debugger/src/utils/wasm.js
@@ -7,6 +7,8 @@
 import { BinaryReader } from "wasmparser/dist/WasmParser";
 import { WasmDisassembler, NameSectionReader } from "wasmparser/dist/WasmDis";
 
+import Services from "devtools-services";
+
 import type { SourceId, WasmSourceContent } from "../types";
 type WasmState = {
   lines: Array<number>,
@@ -154,7 +156,7 @@ export function renderWasmText(
     data[i] = binary.charCodeAt(i);
   }
   const { lines } = getWasmText(sourceId, data);
-  const MAX_LINES = 1000000;
+  const MAX_LINES = Services.prefs.getIntPref("devtools.debugger.wasm-max-lines", 1000000);
   if (lines.length > MAX_LINES) {
     lines.splice(MAX_LINES, lines.length - MAX_LINES);
     lines.push(";; .... text is truncated due to the size");
diff --git a/devtools/client/preferences/debugger.js b/devtools/client/preferences/debugger.js
--- a/devtools/client/preferences/debugger.js
+++ b/devtools/client/preferences/debugger.js
@@ -46,6 +46,7 @@ pref("devtools.debugger.start-panel-coll
 pref("devtools.debugger.end-panel-collapsed", false);
 pref("devtools.debugger.start-panel-size", 300);
 pref("devtools.debugger.end-panel-size", 300);
+pref("devtools.debugger.wasm-max-lines", 1000000);
 pref("devtools.debugger.tabs", "[]");
 pref("devtools.debugger.tabsBlackBoxed", "[]");
 pref("devtools.debugger.pending-selected-location", "{}");
